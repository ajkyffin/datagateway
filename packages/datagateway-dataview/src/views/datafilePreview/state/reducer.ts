import DATAFILE_PREVIEWER_DEFAULT from '../defaults';
import {
  DecrementDatafilePreviewerZoomLevelType,
  IncrementDatafilePreviewerZoomLevelType,
  ResetDatafilePreviewerZoomLevelType,
  ToggleDatafilePreviewerDetailsPanePayload,
  ToggleDatafilePreviewerDetailsPaneType,
} from './actions';
import { DGDataViewState } from '../../../state/app.types';

/**
 * Defines the shape of the redux state for {@link DatafilePreviewer}
 */
interface DatafilePreviewerState {
  isDetailsPaneShown: boolean;
  zoomLevel: number;
}

function toggleDatafilePreviewerDetailsPane(
  state: DGDataViewState,
  payload: ToggleDatafilePreviewerDetailsPanePayload
): DGDataViewState {
  return {
    ...state,
    isisDatafilePreviewer: {
      ...state.isisDatafilePreviewer,
      isDetailsPaneShown: payload.shouldShow,
    },
  };
}

function incDatafilePreviewerZoomLevel(
  state: DGDataViewState
): DGDataViewState {
  return {
    ...state,
    isisDatafilePreviewer: {
      ...state.isisDatafilePreviewer,
      zoomLevel:
        state.isisDatafilePreviewer.zoomLevel +
        DATAFILE_PREVIEWER_DEFAULT.zoomLevelStep,
    },
  };
}

function decDatafilePreviewerZoomLevel(
  state: DGDataViewState
): DGDataViewState {
  return {
    ...state,
    isisDatafilePreviewer: {
      ...state.isisDatafilePreviewer,
      // zoom level should not go lower than zoomLevelStep
      // otherwise will result in zoom level <= 0.
      zoomLevel: Math.max(
        DATAFILE_PREVIEWER_DEFAULT.minZoomLevel,
        state.isisDatafilePreviewer.zoomLevel -
          DATAFILE_PREVIEWER_DEFAULT.zoomLevelStep
      ),
    },
  };
}

function resetDatafilePreviewerZoomLevel(
  state: DGDataViewState
): DGDataViewState {
  return {
    ...state,
    isisDatafilePreviewer: {
      ...state.isisDatafilePreviewer,
      zoomLevel: DATAFILE_PREVIEWER_DEFAULT.zoomLevel,
    },
  };
}

const datafilePreviewerInitialState: DatafilePreviewerState = {
  isDetailsPaneShown: DATAFILE_PREVIEWER_DEFAULT.showDetailsPane,
  zoomLevel: DATAFILE_PREVIEWER_DEFAULT.zoomLevel,
};

const datafilePreviewerReducer = {
  [ToggleDatafilePreviewerDetailsPaneType]: toggleDatafilePreviewerDetailsPane,
  [IncrementDatafilePreviewerZoomLevelType]: incDatafilePreviewerZoomLevel,
  [DecrementDatafilePreviewerZoomLevelType]: decDatafilePreviewerZoomLevel,
  [ResetDatafilePreviewerZoomLevelType]: resetDatafilePreviewerZoomLevel,
};

export {
  datafilePreviewerInitialState,
  datafilePreviewerReducer,
  toggleDatafilePreviewerDetailsPane,
  incDatafilePreviewerZoomLevel,
  decDatafilePreviewerZoomLevel,
  resetDatafilePreviewerZoomLevel,
};
export type { DatafilePreviewerState };
