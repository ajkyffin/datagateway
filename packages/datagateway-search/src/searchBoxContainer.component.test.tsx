import React from 'react';
import { ReactWrapper } from 'enzyme';

import { createShallow } from '@material-ui/core/test-utils';
import SearchBoxContainer from './searchBoxContainer.component';
import SearchBoxContainerSide from './searchBoxContainerSide.component';
import { useSelector } from 'react-redux';
import { initialState } from './state/reducers/dgsearch.reducer';

jest.mock('loglevel');

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}));

describe('SearchBoxContainer - Tests', () => {
  let shallow;

  const testInitiateSearch = jest.fn();

  const createWrapper = (path: string): ReactWrapper => {
    return shallow(<SearchBoxContainer initiateSearch={testInitiateSearch} />);
  };

  beforeEach(() => {
    shallow = createShallow({ untilSelector: 'Grid' });
    useSelector.mockImplementation(() => {
      return initialState;
    });
  });

  it('renders searchBoxContainer correctly', () => {
    const wrapper = createWrapper('/');

    expect(wrapper).toMatchSnapshot();
  });
});

describe('SearchBoxContainerSide - Tests', () => {
  let shallow;

  const testInitiateSearch = jest.fn();

  const createWrapper = (path: string): ReactWrapper => {
    return shallow(
      <SearchBoxContainerSide initiateSearch={testInitiateSearch} />
    );
  };

  beforeEach(() => {
    shallow = createShallow({ untilSelector: 'Grid' });
  });

  it('renders searchBoxContainerSide correctly', () => {
    const wrapper = createWrapper('/');

    expect(wrapper).toMatchSnapshot();
  });
});
