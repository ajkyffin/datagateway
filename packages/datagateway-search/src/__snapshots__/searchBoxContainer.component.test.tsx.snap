// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBoxContainer - Tests renders searchBoxContainer correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-6 makeStyles-containerBox-2"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    direction="row"
    id="container-searchbox"
    justify="center"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      style={
        Object {
          "flexGrow": 1,
        }
      }
      xs="auto"
    >
      <SearchTextBox
        initiateSearch={[MockFunction]}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      style={
        Object {
          "marginTop": "8px",
        }
      }
    >
      <Connect(CheckboxesGroup) />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      style={
        Object {
          "marginTop": "8px",
        }
      }
    >
      <Connect(SelectDates)
        initiateSearch={[MockFunction]}
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      style={
        Object {
          "display": "flex",
          "marginLeft": 6,
          "marginTop": "24px",
        }
      }
    >
      <TranslatedSearchButton
        initiateSearch={[MockFunction]}
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-leftText-3"
    >
      <Trans
        i18nKey="searchBox.examples_label"
        t={[Function]}
      >
        For example
        <WithStyles(ForwardRef(Link))
          href="searchBox.examples_label_link1"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          "instrument calibration"
        </WithStyles(ForwardRef(Link))>
        or
         
        <WithStyles(ForwardRef(Link))
          href="searchBox.examples_label_link2"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          neutron AND scattering
        </WithStyles(ForwardRef(Link))>
        .
      </Trans>
       
      <AdvancedHelpDialogue />
    </WithStyles(ForwardRef(Typography))>
    <div
      className="makeStyles-rightText-4"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexWrap": "nowrap",
          }
        }
      >
        <Memo(ForwardRef(InfoOutlinedIcon))
          className="makeStyles-infoIcon-1"
          fontSize="small"
        />
         
        <WithStyles(ForwardRef(Typography))
          display="inline"
          style={
            Object {
              "fontSize": "14px",
              "paddingLeft": "6px",
            }
          }
        >
          searchBox.limited_results_message {maxNumResults:{tabs:{datasetTab:false,datafileTab:false,investigationTab:false},selectAllSetting:true,settingsLoaded:false,sideLayout:false,searchableEntities:[investigation,dataset,datafile],maxNumResults:300}}
        </WithStyles(ForwardRef(Typography))>
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchBoxContainerSide - Tests renders searchBoxContainerSide correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
  id="container-searchbox"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(MuiBox)
      px={2}
    >
      <SearchTextBox
        initiateSearch={[MockFunction]}
      />
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(MuiBox)
      px={2}
    >
      <Connect(SelectDates)
        initiateSearch={[MockFunction]}
      />
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Connect(CheckboxesGroup) />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Styled(MuiBox)
      mx={5}
      pb={2}
    >
      <TranslatedSearchButton
        initiateSearch={[MockFunction]}
      />
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
